.github/workflows/linux.yml

```diff
@@ -202,7 +202,8 @@ jobs:
             -Db_asneeded=true \
             -Db_lto=true \
             -Dtry_static_libs=png \
-            -Duse_fluidsynth=false \
+            -Dfluidsynth:enable-floats=true \
+            -Dfluidsynth:try-static-deps=true \
```

Comment:
Regarding `-Dfluidsynth:try-static-deps=true`: is this needed to tell the wrap to also build a static `.a` for FluidSynth? 

Ideally FluidSynth's the wrap would provide both shared and static libraries by default (but allow either to be opt-out), not because we from DOSBox Staging want it, but because providing both library types is the most complete. 

I realize this might differ from upstream's cmake behavior - but given they're not interested in taking ownership of the wrap, I see only benefits in addressing some of upstream's cmake deficiency  downstream. 

Infact, I would also be willing to throw enable-flows in by default as well, so users of the FluidSynth wrap benefit from sane default. (using 64-bit doubles for audio samples should be the 0.00001% exception and not the norm).

Reply:
> Regarding -Dfluidsynth:try-static-deps=true: is this needed to tell the wrap to also build a static .a for FluidSynth?

No. `-Ddefault_library=static` (Meson option) switches all wraps to build static libs, but those static libs will be dynamically linked with FluidSynth dependencies.

`-Dfluidsynth:try-static-deps=true` (which is project-specific option, not a Meson option) is needed for attempting linking FluidSynth's dependencies statically (which may or may not fail, we don't know until we try).

Further discussion in here: https://github.com/mesonbuild/meson/issues/7621

> Ideally FluidSynth's wrap would provide both shared and static libraries by default (but allow either to be opt-out), not because we from DOSBox Staging want it, but because providing both library types is the most complete.

It already does :) Meson option is: `--default-library {shared,static,both}`

> Infact, I would also be willing to set enable-floats=true (in the wrap) as well, so users of the FluidSynth wrap benefit from sane and optimized defaults. Given using 64-bit doubles for audio samples should be the rare exception and not the norm.

I am determined to stay in sync with FluidSynth's upstream decisions regarding option naming and default values. However, from the discussion so far I think FluidSynth maintainer agrees with you, so you should probably create discussion/request/PR to change/remove this option upstream - this way all users will benefit :)

Reply:
Thanks for explaining the distinction between the two dynamic options - and the better suggestion regarding FluidSynth floats :-)